plugins/inputs/http_listener/http_listener.go

```diff
@@ -13,135 +13,138 @@ import (
 	"github.com/influxdata/telegraf"
 	"github.com/influxdata/telegraf/internal"
 	"github.com/influxdata/telegraf/plugins/inputs"
-	"github.com/influxdata/telegraf/plugins/inputs/http_listener/stoppableListener"
-	"github.com/influxdata/telegraf/plugins/parsers"
+	"github.com/influxdata/telegraf/plugins/parsers/influx"
+)
+
+const (
+	// DEFAULT_MAX_BODY_SIZE is the default maximum request body size, in bytes.
+	// if the request body is over this size, we will return an HTTP 413 error.
+	// 500 MB
+	DEFAULT_MAX_BODY_SIZE = 500 * 1024 * 1024
+
+	// MAX_LINE_SIZE is the maximum size, in bytes, that can be allocated for
+	// a single InfluxDB point.
+	// 64 KB
+	DEFAULT_MAX_LINE_SIZE = 64 * 1024
 )
 
 type HttpListener struct {
 	ServiceAddress string
 	ReadTimeout    internal.Duration
 	WriteTimeout   internal.Duration
+	MaxBodySize    int64
+	MaxLineSize    int
 
-	sync.Mutex
+	mu sync.Mutex
 	wg sync.WaitGroup
 
-	listener *stoppableListener.StoppableListener
+	listener net.Listener
 
-	parser parsers.Parser
+	parser influx.InfluxParser
 	acc    telegraf.Accumulator
+	pool   *pool
 }
 
 const sampleConfig = `
   ## Address and port to host HTTP listener on
   service_address = ":8186"
 
-  ## timeouts
+  ## maximum duration before timing out read of the request
   read_timeout = "10s"
+  ## maximum duration before timing out write of the response
   write_timeout = "10s"
+
+  ## Maximum allowed http request body size in bytes.
+  ## 0 means to use the default of 536,870,912 bytes (500 mebibytes)
+  max_body_size = 0
+
+  ## Maximum line size allowed to be sent in bytes.
+  ## 0 means to use the default of 65536 bytes (64 kibibytes)
+  max_line_size = 0
 `
 
-func (t *HttpListener) SampleConfig() string {
+func (h *HttpListener) SampleConfig() string {
 	return sampleConfig
 }
 
-func (t *HttpListener) Description() string {
+func (h *HttpListener) Description() string {
 	return "Influx HTTP write listener"
 }
 
-func (t *HttpListener) Gather(_ telegraf.Accumulator) error {
+func (h *HttpListener) Gather(_ telegraf.Accumulator) error {
+	log.Printf("D! The http_listener has created %d buffers", h.pool.ncreated())
 	return nil
 }
 
-func (t *HttpListener) SetParser(parser parsers.Parser) {
-	t.parser = parser
-}
-
 // Start starts the http listener service.
-func (t *HttpListener) Start(acc telegraf.Accumulator) error {
-	t.Lock()
-	defer t.Unlock()
+func (h *HttpListener) Start(acc telegraf.Accumulator) error {
+	h.mu.Lock()
+	defer h.mu.Unlock()
+	h.parser = influx.InfluxParser{}
```

Comment:
This line looks unecessary, unless I'm missing something. Is there some unexported state that needs to be reset if this `HttpListener` is reused?